File: expert_belt_api/models/Record.py
from django.db import models
from django.contrib.auth.models import User

from . import Tournament, UserProfile

# Create your models here.


class Record(models.Model):
    player = models.ForeignKey(User, on_delete=models.RESTRICT, default=None)
    tournament = models.ForeignKey(
        Tournament, on_delete=models.RESTRICT, default=None)

    wins = models.IntegerField(default=None, null=True)
    losses = models.IntegerField(default=None, null=True)
    ties = models.IntegerField(default=None, null=True)
    drop = models.IntegerField(default=None, null=True)
    placing = models.IntegerField(default=None, null=True)

File: frontend/src/API/models/Record.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { TournamentRecord } from './TournamentRecord';

export type Record = {
    player?: number;
    wins?: number | null;
    losses?: number | null;
    ties?: number | null;
    drop?: number | null;
    placing?: number | null;
    tournament: TournamentRecord;
    player_username: string;
};

